keras_cv/layers/regularization/drop_path_test.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+
+from keras_cv.layers import DropPath
+
+
+class DropPathTest(tf.test.TestCase):
+    FEATURE_SHAPE = (1, 14, 14, 256)
+
+    def test_input_unchanged_in_eval_mode(self):
+        layer = DropPath(drop_rate=0.5)
+        inputs = tf.random.uniform(self.FEATURE_SHAPE)
+
+        outputs = layer(inputs, training=False)
+
+        self.assertAllClose(inputs, outputs)
+
+    def test_input_unchanged_with_rate_equal_to_zero(self):
+        layer = DropPath(drop_rate=0)
+        inputs = tf.random.uniform(self.FEATURE_SHAPE)
+
+        outputs = layer(inputs, training=True)
+
+        self.assertAllClose(inputs, outputs)
+
+    def test_input_gets_partially_zeroed_out_in_train_mode(self):
+        layer = DropPath(drop_rate=0.2)
+        inputs = tf.random.uniform(self.FEATURE_SHAPE)
+
+        outputs = layer(inputs, training=True)
+
+        non_zeros_inputs = tf.math.count_nonzero(inputs, dtype=tf.int32)
+        non_zeros_outputs = tf.math.count_nonzero(outputs, dtype=tf.int32)
+
+        self.assertGreaterEqual(non_zeros_inputs, non_zeros_outputs)
+
+    def test_strict_input_gets_partially_zeroed_out_in_train_mode(self):
```

Comment:
Just for your own purposes, it’s good to note to never include probabilistic tests even if they’re extremely likely to pass.

The birthday problem applies to flaky tests.
https://en.wikipedia.org/wiki/Birthday_problem

Reply:
@LukeWood any reason to not do statistical testing (like the one I mentioned [here](https://github.com/keras-team/keras-cv/pull/480/files#r894420636)) of the probabilistic methods like the one added in this PR?

Reply:
As long as you properly seed your layer, the test case, TensorFlow, numpy, and any other library used then no it is not a problem.

Reply:
I am still not too sure why including statistical tests is not a good idea but if you think we're okay without them then we'll go without them. 👍

Reply:
they are totally fine to include - as long as you seed properly. 

The issue is around flaky tests - even if they are incredibly unlikely to flake this becomes a burden at Google scale due to the birthday problem.

Reply:
@sayakpaul In that case could you take a look at the test case and see if it is okay?